tests/infrastructure/instance_types/test_common_vm_preference.py

```
@@ -52,10 +57,20 @@ def start_vm_with_cluster_preference(client, preference_name, namespace_name):
+    """Check if a VirtualMachineClusterPreference requires EFI firmware."""
+    preference = VirtualMachineClusterPreference(client=client, name=preference_name)
+    return bool(preference.instance.spec.get("firmware", {}).get("preferredEfi"))
+
+
 def run_general_vm_preferences(client, namespace, preferences):
+    is_s390x = get_cluster_architecture() == S390X
     for preference_name in preferences:
         # TODO remove arm64 skip when openshift-virtualization-tests support arm64
         if all(suffix not in preference_name for suffix in ["virtio", "arm64"]):
+            if is_s390x and _preference_requires_efi(client=client, preference_name=preference_name):
+                LOGGER.info(f"Skipping preference {preference_name}: EFI/OVMF not available on s390x")
```

Ruff (0.15.7)

[warning] Logging statement uses f-string

(G004)